odoo/models.py

```diff
@@ -1731,19 +1768,326 @@ def clear_caches(cls):
         """
         cls.pool._clear_cache()
 
+    @api.model
+    def _read_group_select(self, query, aggregate_spec):
+        """ return (<SQL expression>, [<fields name used in SQL expression>])"""
+        if aggregate_spec == '__count':
+            return 'COUNT(*)', []
+
+        fname, func = split_read_group_spec(aggregate_spec)
+
+        if fname not in self:
+            raise ValueError(f"Invalid field {fname!r} on model {self._name!r} for {aggregate_spec!r}.")
+        if func not in VALID_AGGREGATE_FUNCTIONS:
+            raise ValueError(f"Invalid aggregate method {func!r} for {aggregate_spec!r}.")
```

Comment:
can we have a more explicit message if no func are given ? (so we know it's not a bug, but it should be explicit)

> Aggregate method is mandatory